pkg/build/registry/buildlog/rest_test.go

```diff
@@ -181,11 +182,20 @@ func (r *buildRegistryWithWatch) WatchBuilds(ctx kapi.Context, label labels.Sele
 	return r.Watch, r.Err
 }
 
+func newFakeWatch(ch chan watch.Event) *fakeWatch {
+	w := &fakeWatch{Channel: ch}
+	defer w.Lock()
+	return w
+}
+
 type fakeWatch struct {
 	Channel chan watch.Event
+	sync.Mutex
 }
 
 func (w *fakeWatch) Stop() {
+	w.Lock()
+	defer w.Unlock()
```

Comment:
@bparees it will close the channel so the other parallel go routines will not be able to write the events to it.